File: solutions/go/word-count/1/word_count.go
package wordcount

import (
	"regexp"
	"strings"
)

type Frequency map[string]int

func WordCount(phrase string) Frequency {
	freq := make(Frequency)

	for _, v := range regexp.MustCompile(`[^\w']+`).Split(phrase, -1) {
		word := strings.Trim(v, "'")
		if word != "" {
			freq[strings.ToLower(word)]++
		}
	}

	return freq
}

File: solutions/typescript/word-count/1/word-count.ts
type Frequency = Map<string, number>

export function count(phrase: string): Frequency {
  const freq: Frequency = new Map()

  for (const [word] of phrase.toLowerCase().matchAll(/\b[\w']+\b/g)) {
    const newValue = (freq.get(word) || 0) + 1
    freq.set(word, newValue)
  }

  return freq
}
